core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -1183,6 +1183,22 @@ impl SubstateDatabase for RocksDBStore {
             .get(&(partition_key.clone(), sort_key.clone()))
     }
 
+    fn list_entries_from(
+        &self,
+        partition_key: &DbPartitionKey,
+        from_sort_key: Option<&DbSortKey>,
+    ) -> Box<dyn Iterator<Item = PartitionEntry> + '_> {
+        let partition_key = partition_key.clone();
+        let from_sort_key = from_sort_key.cloned().unwrap_or(DbSortKey(vec![]));
+        Box::new(
+            self.open_db_context()
+                .cf(SubstatesCf)
+                .iterate_from(&(partition_key.clone(), from_sort_key), Direction::Forward)
```

Comment:
Heads up, @jakrawcz-rdx - with some pretty coincidental foresight, we discussed about the need for this partitioned iteration in the CF trait system yesterday. I guess when this merges we'll need such a trait on the Substates key codec.